File: bot.py
# Imports
import os

import discord
from discord.ext import commands
from dotenv import load_dotenv
from utils.event_functions import (
    add_new_event,
    remove_event,
    format_events,
    events_countdown,
    get_events,
)

# Print current working directory
print(f"Current working directory: {os.getcwd()}")

# Load variables from .env file
load_dotenv()
TOKEN = os.getenv("DISCORD_TOKEN")
if TOKEN is None:
    raise ValueError("TOKEN is not set in environment variables.")

# Set up the bot with proper intents
intents = discord.Intents.default()
intents.message_content = True  # Required for message content
bot = commands.Bot(command_prefix="!", intents=intents)


# Bot events
@bot.event
async def on_ready():
    print(f"Logged in as {bot.user}")
    try:
        synced = await bot.tree.sync()
        print(f"Synced {len(synced)} command(s)")
    except Exception as e:
        print(e)


@bot.tree.command(name="ping", description="Test the bot's latency.")
async def ping(interaction: discord.Interaction):
    await interaction.response.send_message("Pong!")


@bot.tree.command(name="events", description="Display all scheduled events.")
async def events(interaction: discord.Interaction):
    await interaction.response.send_message(format_events(get_events()))


@bot.tree.command(name="add_event", description="Add a new event.")
async def add_event(interaction: discord.Interaction):
    await add_new_event(interaction, bot)


@bot.tree.command(name="delete_event", description="Delete an event.")
async def delete_event(interaction: discord.Interaction):
    await remove_event(interaction, bot)


@bot.tree.command(name="countdown", description="Display the time until all events.")
async def countdown(interaction: discord.Interaction):
    events = get_events()
    await interaction.response.send_message(
        "**Event Countdowns:**\n" + events_countdown(events)
    )


# Run the bot
bot.run(TOKEN)

File: bot.js
// Imports
const { Client, GatewayIntentBits, Partials, Collection } = require('discord.js');
const { config } = require('dotenv');
const eventFunctions = require('./utils/event_functions');

// Load environment variables.
config({ path: 'data/.env' });

// Setup intents.
const client = new Client({
    intents: [
        GatewayIntentBits.Guilds,
        GatewayIntentBits.GuildMessages,
        GatewayIntentBits.MessageContent,
        GatewayIntentBits.GuildMembers,
        GatewayIntentBits.GuildPresences
    ],
    partials: [Partials.Message, Partials.Channel, Partials.Reaction]
});

// Bot Functions
// --- Logged in.
client.once('ready', () => {
    console.log(`Logged in as ${client.user.tag}!`);
});

// --- If the bot is mentioned in message.
client.on('messageCreate', async (message) => {
    // Don't respond to our own messages.
    if (message.author.bot) return;

    // Check if the bot was mentioned in the message.
    if (message.mentions.has(client.user)) {
        // TODO: Fill out command list when pinged.
        await message.channel.send(eventFunctions.help());
    }

    // This line is important! It processes commands.
    client.commands.get(message.content.split(' ')[0].substring(1))?.execute(message);
});

// Commands (like !ping)
client.commands = new Collection();
client.commands.set('ping', {
    name: 'ping',
    description: 'Test to see if bot is working.',
    async execute(message) {
        await message.channel.send('Pong!');
    }
});
client.commands.set('events', {
    name: 'events',
    description: 'Lists help menu for all events functions.',
    async execute(message) {
        await eventFunctions.events(message);
    }
});
client.commands.set('list_events', {
    name: 'list_events',
    description: 'Lists all events in events.json.',
    async execute(message) {
        await eventFunctions.list_events(message);
    }
});
client.commands.set('add_event', {
    name: 'add_event',
    description: 'Add an event to events.json.',
    async execute(message) {
        await eventFunctions.add_event(message);
    }
});
client.commands.set('remove_event', {
    name: 'remove_event',
    description: 'Removes an event from events.json.',
    async execute(message) {
        await eventFunctions.remove_event(message);
    }
});

// Get token from environment variable.
const TOKEN = process.env.DISCORD_TOKEN;
if (!TOKEN) {
    throw new Error("No token found. Make sure DISCORD_TOKEN is set in .env file");
}

client.login(TOKEN);